tests/integration/test.replication.js

```diff
@@ -1548,6 +1548,35 @@ adapters.forEach(function (adapters) {
       });
     });
 
+    it('Empty replication updates checkpoint (#5145)',
+       function (done) {
+      var db = new PouchDB(dbs.name);
+      var remote = new PouchDB(dbs.remote);
+      var docs1 = [ {_id: '0', integer: 0} ];
+      remote.bulkDocs({ docs: docs1 }, function () {
+        var filter = function () {
+          return false;
+        };
+        db.replicate.from(remote, {
+          filter: filter
+        }, function (err, result) {
+          result.ok.should.equal(true);
+          result.docs_written.should.equal(0);
+          result.docs_read.should.equal(0);
+          result.last_seq.should.equal(1);
+          db.replicate.from(remote, {
+            filter: filter
+          }, function (err, result) {
+            result.ok.should.equal(true);
+            result.docs_written.should.equal(0);
+            result.docs_read.should.equal(0);
+            result.last_seq.should.equal(1);
```

Comment:
Yup that is what I suspected, its purely a test problem. Looking at the test I am not sure I understand how it is actually testing this fix, it seems like it should pass either way? Having a hard time thinking of how to actually test this, you could override the ajax function and check that the second replication is triggered with the correct `since`